ratelimit/registry.go

```diff
@@ -19,33 +20,41 @@ const (
 // ratelimiters.
 type Registry struct {
 	sync.Mutex
-	defaults Settings
-	global   Settings
-	lookup   map[Settings]*Ratelimit
-	swarm    Swarmer
+	defaults    Settings
+	global      Settings
+	lookup      map[Settings]*Ratelimit
+	swarm       Swarmer
+	swimOptions *swarm.Options
+	redisRing   *ring
+	quit        chan<- struct{}
 }
 
 // NewRegistry initializes a registry with the provided default settings.
 func NewRegistry(settings ...Settings) *Registry {
-	return NewSwarmRegistry(nil, settings...)
+	return NewSwarmRegistry(nil, nil, nil, settings...)
 }
 
 // NewSwarmRegistry initializes a registry with an optional swarm and
 // the provided default settings. If swarm is nil, clusterRatelimits
 // will be replaced by voidRatelimit, which is a noop limiter implementation.
-func NewSwarmRegistry(swarm Swarmer, settings ...Settings) *Registry {
+func NewSwarmRegistry(swarm Swarmer, swimOptions *swarm.Options, ro *RedisOptions, settings ...Settings) *Registry {
 	defaults := Settings{
 		Type:          DisableRatelimit,
 		MaxHits:       DefaultMaxhits,
 		TimeWindow:    DefaultTimeWindow,
 		CleanInterval: DefaultCleanInterval,
 	}
 
+	q := make(chan struct{})
+
 	r := &Registry{
-		defaults: defaults,
-		global:   defaults,
-		lookup:   make(map[Settings]*Ratelimit),
-		swarm:    swarm,
+		defaults:    defaults,
+		global:      defaults,
+		lookup:      make(map[Settings]*Ratelimit),
+		swarm:       swarm,
+		swimOptions: swimOptions,
```

Comment:
the registry doesn't need the *swarm.Options field, it is enough if it knows that there is a swarm instance or not.